### PR TITLE
msmtp: Fix build for Linux

### LINKED ISSUE #13134: Cannot upgrade msmtp to 1.8.4
**Please note we will close your issue without comment if you delete, do not read or do not fill out the issue checklist below and provide ALL the requested information. If you repeatedly fail to use the issue template, we will block you from ever submitting issues to Homebrew again.**

- [x] are reporting a bug others will be able to reproduce and not asking a question or requesting software. If you're not sure or want to ask a question do so on our Discourse: https://discourse.brew.sh. To get software added or changed in Homebrew please file a [Pull Request](https://github.com/Homebrew/linuxbrew-core/blob/master/CONTRIBUTING.md)
- [x] have a problem with `brew install` (or `upgrade`, `reinstall`) a single, Homebrew/homebrew-core formula (not cask) on macOS? If it's a general `brew` problem please file this issue at Homebrew/brew: https://github.com/Homebrew/brew/issues/new/choose. If it's a Linux problem please file this issue at https://github.com/Homebrew/linuxbrew-core/issues/new/choose. If it's a `brew cask` problem please file this issue at https://github.com/Homebrew/homebrew-cask/issues/new/choose. If it's a tap (e.g. Homebrew/homebrew-php) problem please file this issue at the tap.
- [x] ran `brew update` and can still reproduce the problem?
- [x] ran `brew doctor`, fixed all issues and can still reproduce the problem?
- [x] ran `brew gist-logs <formula>` (where `<formula>` is the name of the formula that failed) and included the output link?

## Gist

[https://gist.github.com/CaptainQuirk/c7a1e0e8d51285ccb624766fda3f31d6](https://gist.github.com/CaptainQuirk/c7a1e0e8d51285ccb624766fda3f31d6)

## What you were trying to do (and why)

I was trying to upgrade msmtp from **1.6.6** to **1.8.4**

## What happened (include command output)

```
==> Auto-updated Homebrew!
Updated 1 tap (homebrew/core).
No changes to formulae.

==> Upgrading 1 outdated package:
msmtp 1.6.6 -> 1.8.4
==> Upgrading msmtp 
==> Downloading https://marlam.de/msmtp/releases/msmtp-1.8.4.tar.xz
Already downloaded: /home/leonard/.cache/Homebrew/downloads/64660a81c08672d9d55d622495beaac312e3fecce1223f894958b47d3f126e70--msmtp-1.8.4.tar.xz
==> ./configure --disable-silent-rules --with-macosx-keyring --prefix=/home/linuxbrew/.linuxbrew/Cellar/msmtp/1.8.4
==> make install
Last 15 lines from /home/leonard/.cache/Homebrew/Logs/msmtp/02.make:
gcc-5 -DHAVE_CONFIG_H -DSYSCONFDIR=\"/home/linuxbrew/.linuxbrew/Cellar/msmtp/1.8.4/etc\" -DLOCALEDIR=\"/home/linuxbrew/.linuxbrew/Cellar/msmtp/1.8.4/share/locale\" -DBINDIR=\"/home/linuxbrew/.linuxbrew/Cellar/msmtp/1.8.4/bin\" -I. -I..  -I/home/linuxbrew/.linuxbrew/Cellar/libtasn1/4.13/include -I/home/linuxbrew/.linuxbrew/Cellar/nettle/3.4.1/include -I/home/linuxbrew/.linuxbrew/Cellar/p11-kit/0.23.15/include/p11-kit-1 -I/home/linuxbrew/.linuxbrew/Cellar/gnutls/3.6.7.1/include   -pthread -I/home/linuxbrew/.linuxbrew/Cellar/libffi/3.2.1/lib/libffi-3.2.1/include -I/home/linuxbrew/.linuxbrew/Cellar/zlib/1.2.11/include -I/home/linuxbrew/.linuxbrew/Cellar/libsecret/0.18.7/include/libsecret-1 -I/home/linuxbrew/.linuxbrew/Cellar/glib/2.60.2/include -I/home/linuxbrew/.linuxbrew/Cellar/util-linux/2.33.2_1/include/libmount -I/home/linuxbrew/.linuxbrew/Cellar/util-linux/2.33.2_1/include/blkid -I/home/linuxbrew/.linuxbrew/Cellar/util-linux/2.33.2_1/include/uuid -I/home/linuxbrew/.linuxbrew/Cellar/glib/2.60.2/include -I/home/linuxbrew/.linuxbrew/Cellar/glib/2.60.2/include/glib-2.0 -I/home/linuxbrew/.linuxbrew/Cellar/glib/2.60.2/lib/glib-2.0/include -I/home/linuxbrew/.linuxbrew/opt/gettext/include -I/home/linuxbrew/.linuxbrew/Cellar/pcre/8.43/include   -g -O2 -c -o smtp.o smtp.c
gcc-5 -DHAVE_CONFIG_H -DSYSCONFDIR=\"/home/linuxbrew/.linuxbrew/Cellar/msmtp/1.8.4/etc\" -DLOCALEDIR=\"/home/linuxbrew/.linuxbrew/Cellar/msmtp/1.8.4/share/locale\" -DBINDIR=\"/home/linuxbrew/.linuxbrew/Cellar/msmtp/1.8.4/bin\" -I. -I..  -I/home/linuxbrew/.linuxbrew/Cellar/libtasn1/4.13/include -I/home/linuxbrew/.linuxbrew/Cellar/nettle/3.4.1/include -I/home/linuxbrew/.linuxbrew/Cellar/p11-kit/0.23.15/include/p11-kit-1 -I/home/linuxbrew/.linuxbrew/Cellar/gnutls/3.6.7.1/include   -pthread -I/home/linuxbrew/.linuxbrew/Cellar/libffi/3.2.1/lib/libffi-3.2.1/include -I/home/linuxbrew/.linuxbrew/Cellar/zlib/1.2.11/include -I/home/linuxbrew/.linuxbrew/Cellar/libsecret/0.18.7/include/libsecret-1 -I/home/linuxbrew/.linuxbrew/Cellar/glib/2.60.2/include -I/home/linuxbrew/.linuxbrew/Cellar/util-linux/2.33.2_1/include/libmount -I/home/linuxbrew/.linuxbrew/Cellar/util-linux/2.33.2_1/include/blkid -I/home/linuxbrew/.linuxbrew/Cellar/util-linux/2.33.2_1/include/uuid -I/home/linuxbrew/.linuxbrew/Cellar/glib/2.60.2/include -I/home/linuxbrew/.linuxbrew/Cellar/glib/2.60.2/include/glib-2.0 -I/home/linuxbrew/.linuxbrew/Cellar/glib/2.60.2/lib/glib-2.0/include -I/home/linuxbrew/.linuxbrew/opt/gettext/include -I/home/linuxbrew/.linuxbrew/Cellar/pcre/8.43/include   -g -O2 -c -o stream.o stream.c
gcc-5 -DHAVE_CONFIG_H -DSYSCONFDIR=\"/home/linuxbrew/.linuxbrew/Cellar/msmtp/1.8.4/etc\" -DLOCALEDIR=\"/home/linuxbrew/.linuxbrew/Cellar/msmtp/1.8.4/share/locale\" -DBINDIR=\"/home/linuxbrew/.linuxbrew/Cellar/msmtp/1.8.4/bin\" -I. -I..  -I/home/linuxbrew/.linuxbrew/Cellar/libtasn1/4.13/include -I/home/linuxbrew/.linuxbrew/Cellar/nettle/3.4.1/include -I/home/linuxbrew/.linuxbrew/Cellar/p11-kit/0.23.15/include/p11-kit-1 -I/home/linuxbrew/.linuxbrew/Cellar/gnutls/3.6.7.1/include   -pthread -I/home/linuxbrew/.linuxbrew/Cellar/libffi/3.2.1/lib/libffi-3.2.1/include -I/home/linuxbrew/.linuxbrew/Cellar/zlib/1.2.11/include -I/home/linuxbrew/.linuxbrew/Cellar/libsecret/0.18.7/include/libsecret-1 -I/home/linuxbrew/.linuxbrew/Cellar/glib/2.60.2/include -I/home/linuxbrew/.linuxbrew/Cellar/util-linux/2.33.2_1/include/libmount -I/home/linuxbrew/.linuxbrew/Cellar/util-linux/2.33.2_1/include/blkid -I/home/linuxbrew/.linuxbrew/Cellar/util-linux/2.33.2_1/include/uuid -I/home/linuxbrew/.linuxbrew/Cellar/glib/2.60.2/include -I/home/linuxbrew/.linuxbrew/Cellar/glib/2.60.2/include/glib-2.0 -I/home/linuxbrew/.linuxbrew/Cellar/glib/2.60.2/lib/glib-2.0/include -I/home/linuxbrew/.linuxbrew/opt/gettext/include -I/home/linuxbrew/.linuxbrew/Cellar/pcre/8.43/include   -g -O2 -c -o tools.o tools.c
gcc-5 -DHAVE_CONFIG_H -DSYSCONFDIR=\"/home/linuxbrew/.linuxbrew/Cellar/msmtp/1.8.4/etc\" -DLOCALEDIR=\"/home/linuxbrew/.linuxbrew/Cellar/msmtp/1.8.4/share/locale\" -DBINDIR=\"/home/linuxbrew/.linuxbrew/Cellar/msmtp/1.8.4/bin\" -I. -I..  -I/home/linuxbrew/.linuxbrew/Cellar/libtasn1/4.13/include -I/home/linuxbrew/.linuxbrew/Cellar/nettle/3.4.1/include -I/home/linuxbrew/.linuxbrew/Cellar/p11-kit/0.23.15/include/p11-kit-1 -I/home/linuxbrew/.linuxbrew/Cellar/gnutls/3.6.7.1/include   -pthread -I/home/linuxbrew/.linuxbrew/Cellar/libffi/3.2.1/lib/libffi-3.2.1/include -I/home/linuxbrew/.linuxbrew/Cellar/zlib/1.2.11/include -I/home/linuxbrew/.linuxbrew/Cellar/libsecret/0.18.7/include/libsecret-1 -I/home/linuxbrew/.linuxbrew/Cellar/glib/2.60.2/include -I/home/linuxbrew/.linuxbrew/Cellar/util-linux/2.33.2_1/include/libmount -I/home/linuxbrew/.linuxbrew/Cellar/util-linux/2.33.2_1/include/blkid -I/home/linuxbrew/.linuxbrew/Cellar/util-linux/2.33.2_1/include/uuid -I/home/linuxbrew/.linuxbrew/Cellar/glib/2.60.2/include -I/home/linuxbrew/.linuxbrew/Cellar/glib/2.60.2/include/glib-2.0 -I/home/linuxbrew/.linuxbrew/Cellar/glib/2.60.2/lib/glib-2.0/include -I/home/linuxbrew/.linuxbrew/opt/gettext/include -I/home/linuxbrew/.linuxbrew/Cellar/pcre/8.43/include   -g -O2 -c -o xalloc.o xalloc.c
gcc-5 -DHAVE_CONFIG_H -DSYSCONFDIR=\"/home/linuxbrew/.linuxbrew/Cellar/msmtp/1.8.4/etc\" -DLOCALEDIR=\"/home/linuxbrew/.linuxbrew/Cellar/msmtp/1.8.4/share/locale\" -DBINDIR=\"/home/linuxbrew/.linuxbrew/Cellar/msmtp/1.8.4/bin\" -I. -I..  -I/home/linuxbrew/.linuxbrew/Cellar/libtasn1/4.13/include -I/home/linuxbrew/.linuxbrew/Cellar/nettle/3.4.1/include -I/home/linuxbrew/.linuxbrew/Cellar/p11-kit/0.23.15/include/p11-kit-1 -I/home/linuxbrew/.linuxbrew/Cellar/gnutls/3.6.7.1/include   -pthread -I/home/linuxbrew/.linuxbrew/Cellar/libffi/3.2.1/lib/libffi-3.2.1/include -I/home/linuxbrew/.linuxbrew/Cellar/zlib/1.2.11/include -I/home/linuxbrew/.linuxbrew/Cellar/libsecret/0.18.7/include/libsecret-1 -I/home/linuxbrew/.linuxbrew/Cellar/glib/2.60.2/include -I/home/linuxbrew/.linuxbrew/Cellar/util-linux/2.33.2_1/include/libmount -I/home/linuxbrew/.linuxbrew/Cellar/util-linux/2.33.2_1/include/blkid -I/home/linuxbrew/.linuxbrew/Cellar/util-linux/2.33.2_1/include/uuid -I/home/linuxbrew/.linuxbrew/Cellar/glib/2.60.2/include -I/home/linuxbrew/.linuxbrew/Cellar/glib/2.60.2/include/glib-2.0 -I/home/linuxbrew/.linuxbrew/Cellar/glib/2.60.2/lib/glib-2.0/include -I/home/linuxbrew/.linuxbrew/opt/gettext/include -I/home/linuxbrew/.linuxbrew/Cellar/pcre/8.43/include   -g -O2 -c -o aliases.o aliases.c
gcc-5 -DHAVE_CONFIG_H -DSYSCONFDIR=\"/home/linuxbrew/.linuxbrew/Cellar/msmtp/1.8.4/etc\" -DLOCALEDIR=\"/home/linuxbrew/.linuxbrew/Cellar/msmtp/1.8.4/share/locale\" -DBINDIR=\"/home/linuxbrew/.linuxbrew/Cellar/msmtp/1.8.4/bin\" -I. -I..  -I/home/linuxbrew/.linuxbrew/Cellar/libtasn1/4.13/include -I/home/linuxbrew/.linuxbrew/Cellar/nettle/3.4.1/include -I/home/linuxbrew/.linuxbrew/Cellar/p11-kit/0.23.15/include/p11-kit-1 -I/home/linuxbrew/.linuxbrew/Cellar/gnutls/3.6.7.1/include   -pthread -I/home/linuxbrew/.linuxbrew/Cellar/libffi/3.2.1/lib/libffi-3.2.1/include -I/home/linuxbrew/.linuxbrew/Cellar/zlib/1.2.11/include -I/home/linuxbrew/.linuxbrew/Cellar/libsecret/0.18.7/include/libsecret-1 -I/home/linuxbrew/.linuxbrew/Cellar/glib/2.60.2/include -I/home/linuxbrew/.linuxbrew/Cellar/util-linux/2.33.2_1/include/libmount -I/home/linuxbrew/.linuxbrew/Cellar/util-linux/2.33.2_1/include/blkid -I/home/linuxbrew/.linuxbrew/Cellar/util-linux/2.33.2_1/include/uuid -I/home/linuxbrew/.linuxbrew/Cellar/glib/2.60.2/include -I/home/linuxbrew/.linuxbrew/Cellar/glib/2.60.2/include/glib-2.0 -I/home/linuxbrew/.linuxbrew/Cellar/glib/2.60.2/lib/glib-2.0/include -I/home/linuxbrew/.linuxbrew/opt/gettext/include -I/home/linuxbrew/.linuxbrew/Cellar/pcre/8.43/include   -g -O2 -c -o password.o password.c
gcc-5 -DHAVE_CONFIG_H -DSYSCONFDIR=\"/home/linuxbrew/.linuxbrew/Cellar/msmtp/1.8.4/etc\" -DLOCALEDIR=\"/home/linuxbrew/.linuxbrew/Cellar/msmtp/1.8.4/share/locale\" -DBINDIR=\"/home/linuxbrew/.linuxbrew/Cellar/msmtp/1.8.4/bin\" -I. -I..  -I/home/linuxbrew/.linuxbrew/Cellar/libtasn1/4.13/include -I/home/linuxbrew/.linuxbrew/Cellar/nettle/3.4.1/include -I/home/linuxbrew/.linuxbrew/Cellar/p11-kit/0.23.15/include/p11-kit-1 -I/home/linuxbrew/.linuxbrew/Cellar/gnutls/3.6.7.1/include   -pthread -I/home/linuxbrew/.linuxbrew/Cellar/libffi/3.2.1/lib/libffi-3.2.1/include -I/home/linuxbrew/.linuxbrew/Cellar/zlib/1.2.11/include -I/home/linuxbrew/.linuxbrew/Cellar/libsecret/0.18.7/include/libsecret-1 -I/home/linuxbrew/.linuxbrew/Cellar/glib/2.60.2/include -I/home/linuxbrew/.linuxbrew/Cellar/util-linux/2.33.2_1/include/libmount -I/home/linuxbrew/.linuxbrew/Cellar/util-linux/2.33.2_1/include/blkid -I/home/linuxbrew/.linuxbrew/Cellar/util-linux/2.33.2_1/include/uuid -I/home/linuxbrew/.linuxbrew/Cellar/glib/2.60.2/include -I/home/linuxbrew/.linuxbrew/Cellar/glib/2.60.2/include/glib-2.0 -I/home/linuxbrew/.linuxbrew/Cellar/glib/2.60.2/lib/glib-2.0/include -I/home/linuxbrew/.linuxbrew/opt/gettext/include -I/home/linuxbrew/.linuxbrew/Cellar/pcre/8.43/include   -g -O2 -c -o tls.o tls.c
password.c:31:31: fatal error: libsecret/secret.h: No such file or directory
compilation terminated.
Makefile:457: recipe for target 'password.o' failed
make[1]: *** [password.o] Error 1
make[1]: *** Waiting for unfinished jobs....
make[1]: Leaving directory '/tmp/msmtp-20190516-3220-1cvfzzc/msmtp-1.8.4/src'
Makefile:409: recipe for target 'install-recursive' failed
make: *** [install-recursive] Error 1

READ THIS: https://docs.brew.sh/Troubleshooting
```

## What you expected to happen

Msmtp to be upgraded

## Step-by-step reproduction instructions (by running `brew install` commands)

- `$ brew install msmtp`

### PR DESCRIPTION
**Pull request by issyl0**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/linuxbrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/linuxbrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?
- [x] Have you included the output of `brew gist-logs <formula>` of the build failure if your PR fixes a build failure. Please quote the exact error message.

-----

- Fixes #13134.
- Gist logs from issue 13134: https://gist.github.com/CaptainQuirk/c7a1e0e8d51285ccb624766fda3f31d6
- This was failing to find the `libsecret` headers.
- Now use the new `uses_from_macos` syntax to specify dependencies to be
  installed on Linux.

### CODE REVIEW

File: Formula/msmtp.rb [hunk 1]
```diff
@@ -5,14 +5,15 @@ class Msmtp < Formula
   sha256 "e5dd7fe95bc8e2f5eea3e4894ec9628252f30bd700a7fd1a568b10efa91129f7"
 
   bottle do
-    cellar :any
+    cellar :any_skip_relocation
```
Comment (sjackman) [inline Formula/msmtp.rb]: Please revert this change.

Comment (issyl0) [reply Formula/msmtp.rb]: Ah, OK. I based this change off the fix to a similar CI failure in https://github.com/Homebrew/linuxbrew-core/pull/13684#issuecomment-502469937.

Comment (issyl0) [reply Formula/msmtp.rb]: Reverted.

**Comment by sjackman** [comment]
```
Error: --keep-old was passed but there are changes in:
cellar: old: :any, new: "/home/linuxbrew/.linuxbrew/Cellar"
```
https://circleci.com/gh/Homebrew/linuxbrew-core/30997

You could troubleshoot the root cause of this issue (which can be interesting some times), or the workaround (which is fine) is to delete the line `cellar :any`.

**Comment by issyl0** [comment]
> You could troubleshoot the root cause of this issue

You'd have to tell me where to start, please. :wink: In the meantime, I'll delete the line. :wink:

**Comment by sjackman** [comment]
```
$ brew bottle --verbose --json msmtp --keep-old
…
Warning: String '/home/linuxbrew/.linuxbrew/Cellar' still exists in these files:
/home/linuxbrew/.linuxbrew/Cellar/msmtp/1.8.4/bin/msmtp
 --> match '/home/linuxbrew/.linuxbrew/Cellar/msmtp/1.8.4/share/locale' at offset 0x17830
```
https://circleci.com/gh/Homebrew/linuxbrew-core/30997

The bottle on Linux has this reference in a binary file to an absolute path under `/home/linuxbrew`, which on Linux makes the bottle non-relocatable. On macOS it evidently does not have this reference. This is most likely a rabbit hole that you do not need to pursue.

**Comment by iMichka** [comment]
We have a bunch of bottles which are impacted by excatly the same issue with the `locale` folder. Definitvely something which we need to have a look at.

**Comment by sjackman** [comment]
I'm super curious to know why this issue with the `locale` folder does not appear to affect macOS.
